src/ut/utils_test.cpp

```diff
@@ -172,3 +172,103 @@ TEST_F(UtilsTest, RemoveBracketsFromBareIPv6Address)
 {
   EXPECT_EQ("::1", remove_brackets_from_ip("::1"));
 }
+
+TEST_F(UtilsTest, IsAWildcard)
+{
+  EXPECT_TRUE(is_wildcard_uri("sip:!.!*!@domain"));
```

Comment:
These are so cheap to write that I think it would be worth adding a few more `EXPECT_TRUE` statements to this test, specifically testing some of the following things...
* user part doesn't start with `!`
* user part doesn't finish with `!`
* More `!`
* Include other regex characters

I'm not sure how valuable those last two things are; it feels like they might be valuable if we changed the implementation of `is_wildcard_uri` ever.